docs/content/2.usage/1.config.md
---
title: Configuration
description: Detailed description of available module options.

navigation:
  icon: i-lucide-cog
---

## Default configuration

By default, the module uses the following configuration values that you can adjust:

```typescript [nuxt.config.ts]
export default defineNuxtConfig({
  modules: ['nuxt-laravel-echo'],

  echo: {
    broadcaster: 'reverb', // available: reverb, pusher
    host: 'localhost',
    port: 8080,
    scheme: 'https', // available: http, https
    transports: ['ws', 'wss'],
    authentication: {
      mode: 'cookie',
      baseUrl: 'http://localhost:80',
      authEndpoint: '/broadcasting/auth',
      csrfEndpoint: '/sanctum/csrf-cookie',
      csrfCookie: 'XSRF-TOKEN',
      csrfHeader: 'X-XSRF-TOKEN',
    },
    logLevel: 3,
    properties: undefined,
  },
})
```

If you don't specify any value in your `nuxt.config.ts`, then the default config will be used on plugin initialization.

## Overrides

Module configuration is exposed to `runtimeConfig` property of your Nuxt app, 
so you can override either in `echo` module config or `runtimeConfig.public.echo` property.

```typescript [nuxt.config.ts]
export default defineNuxtConfig({
  modules: ['nuxt-laravel-echo'],
  
  echo: {
    key: 'MY_APP_KEY',
  },

  runtimeConfig: {
    public: {
      echo: {
        logLevel: 3,
      },
    },
  },
})
```

## Environment variables

It is possible to override options via environment variables too. 
It might be useful when you want to use `.env` file to provide key for the broadcast backend.

And here is what it will look like in `.env` file:

```env [.env]
NUXT_PUBLIC_ECHO_KEY='REPLACE_ME'
```

## Available options

For any additional configurations, you can adjust the next list of available parameters:

| Parameter | Description | Default |
| --------- | ----------- | ------- |
| **key** | The Laravel Broadcasting app key for a secure connection. | `undefined` |
| **broadcaster** | The Laravel broadcaster type to use. Use reverb or pusher. | `reverb` |
| **host** | The host to connect to WebSocket. | `localhost` |
| **port** | The port to connect to WebSocket. | `8080` |
| **scheme** | The scheme to use for the connection. Use http or https. | `https` |
| **cluster** | The application cluster to connect to. | `undefined` |
| **transports** | The transports to enable for the connection. | `['ws', 'wss']` |
| **authentication** | Authentication options for Private and Presence channels. | `object` |
| **authentication.mode** | Authentication mode cookie or token | `cookie` |
| **authentication.baseUrl** | The base URL of Laravel application. | `http://localhost:80` |
| **authentication.authEndpoint** | The endpoint used for channels authentication. | `/broadcasting/auth` |
| **authentication.csrfEndpoint** | The endpoint used for CSRF token retrieval. | `/sanctum/csrf-cookie` |
| **authentication.csrfCookie** | The name of the CSRF cookie. | `XSRF-TOKEN` |
| **authentication.csrfHeader** | The name of the CSRF header. | `X-XSRF-TOKEN` |
| **logLevel** | The log level to use for the logger. | `3` |
| **properties** | Additional properties to extend the Echo instance options. See Additional properties section. | `undefined` |

## Additional properties

When you are using the Pusher backend, you may need to assign additional properties to your Echo instance object such as:

- enableStats
- activityTimeout
- etc

For these, you can override `echo.properties` parameter to pass all the details that are not included in the default config:

```typescript [nuxt.config.ts]
export default defineNuxtConfig({
  modules: ['nuxt-laravel-echo'],

  echo: {
    key: 'REPLACE_ME',
    properties: {
      customField: 'customValue',
    },
  },
})
```


docs/content/3.composables/2.useEchoConfig.md
---
title: useEchoConfig
description: Accessing the current Echo configuration from your Nuxt app. 
navigation:
  icon: i-lucide-parentheses
---

## Usage

This composable provides quick access to Nuxt configuration under the `echo` key. 
It is used as a part of the module implementation, so there is no particular need to use this in your application.

## Code sample

Here is an example of the call:

```typescript [SomeService.ts]
const echoConfig = useEchoConfig()

console.log(echoConfig.host) // print the broadcasting host URL
```



CHANGELOG.md
# Changelog


## v0.2.8

[compare changes](https://github.com/manchenkoff/nuxt-laravel-echo/compare/v0.2.7...v0.2.8)

### 🏡 Chore

- **deps-dev:** Bump nuxt from 4.0.1 to 4.0.2 ([fe32085](https://github.com/manchenkoff/nuxt-laravel-echo/commit/fe32085))
- **deps-dev:** Bump @nuxt/eslint-config from 1.7.0 to 1.7.1 ([b0485fa](https://github.com/manchenkoff/nuxt-laravel-echo/commit/b0485fa))
- **deps-dev:** Bump eslint from 9.31.0 to 9.32.0 ([2ef1f14](https://github.com/manchenkoff/nuxt-laravel-echo/commit/2ef1f14))
- **deps-dev:** Bump @nuxt/kit from 4.0.1 to 4.0.2 ([bc030a4](https://github.com/manchenkoff/nuxt-laravel-echo/commit/bc030a4))
- **deps-dev:** Bump @nuxt/module-builder from 1.0.1 to 1.0.2 ([843fcae](https://github.com/manchenkoff/nuxt-laravel-echo/commit/843fcae))
- **deps-dev:** Bump nuxt in the npm_and_yarn group across 1 directory ([0d5bcd5](https://github.com/manchenkoff/nuxt-laravel-echo/commit/0d5bcd5))

### 🤖 CI

- Simplify github actions ([f4e9eea](https://github.com/manchenkoff/nuxt-laravel-echo/commit/f4e9eea))

### ❤️ Contributors

- Manchenkoff ([@manchenkoff](https://github.com/manchenkoff))

## v0.2.7

[compare changes](https://github.com/manchenkoff/nuxt-laravel-echo/compare/v0.2.6...v0.2.7)

### 🩹 Fixes

- Bump packages and fixed vulnerability ([a04d7dd](https://github.com/manchenkoff/nuxt-laravel-echo/commit/a04d7dd))

### ❤️ Contributors

- Manchenkoff ([@manchenkoff](https://github.com/manchenkoff))

## v0.2.6

[compare changes](https://github.com/manchenkoff/nuxt-laravel-echo/compare/v0.2.5...v0.2.6)

### 🏡 Chore

- **deps-dev:** Bump @nuxt/kit from 3.17.3 to 3.17.4 ([2008bc6](https://github.com/manchenkoff/nuxt-laravel-echo/commit/2008bc6))
- **deps-dev:** Bump @types/node from 22.15.17 to 22.15.29 ([13de480](https://github.com/manchenkoff/nuxt-laravel-echo/commit/13de480))
- **deps-dev:** Bump @nuxt/eslint-config from 1.3.1 to 1.4.1 ([76c9efc](https://github.com/manchenkoff/nuxt-laravel-echo/commit/76c9efc))
- **deps-dev:** Bump @nuxt/test-utils from 3.18.0 to 3.19.1 ([11d200a](https://github.com/manchenkoff/nuxt-laravel-echo/commit/11d200a))
- **deps-dev:** Bump nuxt from 3.17.3 to 3.17.5 ([d2714ed](https://github.com/manchenkoff/nuxt-laravel-echo/commit/d2714ed))
- **deps-dev:** Bump vitest from 3.1.3 to 3.2.4 ([75cc25e](https://github.com/manchenkoff/nuxt-laravel-echo/commit/75cc25e))
- **deps-dev:** Bump @types/node from 22.15.29 to 24.0.8 ([46dcf25](https://github.com/manchenkoff/nuxt-laravel-echo/commit/46dcf25))
- **deps-dev:** Bump @nuxt/devtools from 2.4.0 to 2.6.0 ([43fa262](https://github.com/manchenkoff/nuxt-laravel-echo/commit/43fa262))
- **deps-dev:** Bump eslint from 9.26.0 to 9.30.0 ([a06a8ca](https://github.com/manchenkoff/nuxt-laravel-echo/commit/a06a8ca))
- Bump packages and apply formatting ([8a0d8c4](https://github.com/manchenkoff/nuxt-laravel-echo/commit/8a0d8c4))

### ❤️ Contributors

- Manchenkoff ([@manchenkoff](https://github.com/manchenkoff))

## v0.2.5

[compare changes](https://github.com/manchenkoff/nuxt-laravel-echo/compare/v0.2.4...v0.2.5)

### 🩹 Fixes

- Throw error on unsupported broadcaster ([6928940](https://github.com/manchenkoff/nuxt-laravel-echo/commit/6928940))

### 🏡 Chore

- **deps-dev:** Bump @nuxt/kit from 3.16.2 to 3.17.1 ([ff692ff](https://github.com/manchenkoff/nuxt-laravel-echo/commit/ff692ff))
- **deps-dev:** Bump nuxt from 3.16.2 to 3.17.1 ([31c4b4c](https://github.com/manchenkoff/nuxt-laravel-echo/commit/31c4b4c))
- **deps-dev:** Bump @nuxt/devtools from 2.3.2 to 2.4.0 ([fe5eb2c](https://github.com/manchenkoff/nuxt-laravel-echo/commit/fe5eb2c))
- **deps-dev:** Bump @nuxt/module-builder from 0.8.4 to 1.0.1 ([71607cb](https://github.com/manchenkoff/nuxt-laravel-echo/commit/71607cb))
- **deps-dev:** Bump vue-tsc from 2.2.8 to 2.2.10 ([2bfdca8](https://github.com/manchenkoff/nuxt-laravel-echo/commit/2bfdca8))

### 🤖 CI

- Added github action permissions ([4751880](https://github.com/manchenkoff/nuxt-laravel-echo/commit/4751880))

### ❤️ Contributors

- Manchenkoff ([@manchenkoff](https://github.com/manchenkoff))

## v0.2.4

[compare changes](https://github.com/manchenkoff/nuxt-laravel-echo/compare/v0.2.3...v0.2.4)

### 🚀 Enhancements

- **interceptors:** Added support for custom user-defined ofetch interceptors ([fbe90b7](https://github.com/manchenkoff/nuxt-laravel-echo/commit/fbe90b7))

### 🏡 Chore

- **deps-dev:** Bump vite from 6.2.3 to 6.2.4 in the npm_and_yarn group ([f60c179](https://github.com/manchenkoff/nuxt-laravel-echo/commit/f60c179))
- **deps-dev:** Bump @nuxt/devtools from 2.3.1 to 2.3.2 ([b45ca4b](https://github.com/manchenkoff/nuxt-laravel-echo/commit/b45ca4b))
- **deps-dev:** Bump @nuxt/eslint-config from 1.2.0 to 1.3.0 ([5bd39f6](https://github.com/manchenkoff/nuxt-laravel-echo/commit/5bd39f6))
- **deps-dev:** Bump @types/node from 22.13.13 to 22.13.16 ([30dccd9](https://github.com/manchenkoff/nuxt-laravel-echo/commit/30dccd9))
- **deps-dev:** Bump nuxt from 3.16.1 to 3.16.2 ([31bc195](https://github.com/manchenkoff/nuxt-laravel-echo/commit/31bc195))
- **deps-dev:** Bump @nuxt/kit from 3.16.1 to 3.16.2 ([1d47692](https://github.com/manchenkoff/nuxt-laravel-echo/commit/1d47692))
- **deps-dev:** Bump vite from 6.2.4 to 6.2.5 in the npm_and_yarn group ([5654b5c](https://github.com/manchenkoff/nuxt-laravel-echo/commit/5654b5c))
- **playground:** Added example of echo interceptor ([5829aed](https://github.com/manchenkoff/nuxt-laravel-echo/commit/5829aed))

### ❤️ Contributors

- Manchenkoff ([@manchenkoff](https://github.com/manchenkoff))

## v0.2.3

[compare changes](https://github.com/manchenkoff/nuxt-laravel-echo/compare/v0.2.2...v0.2.3)

### 🏡 Chore

- Upgraded nuxt packages ([6f5fd46](https://github.com/manchenkoff/nuxt-laravel-echo/commit/6f5fd46))
- Override for esbuild vulnerability ([21660b1](https://github.com/manchenkoff/nuxt-laravel-echo/commit/21660b1))
- Removed override for esbuild ([e2d2cfc](https://github.com/manchenkoff/nuxt-laravel-echo/commit/e2d2cfc))

### ❤️ Contributors


src/runtime/types/options.ts
import type { Broadcaster } from 'laravel-echo'

export interface Authentication {
  /**
   * Authentication mode 'cookie' or 'token'
   * @default 'cookie'
   */
  mode: 'cookie' | 'token'
  /**
   * The base URL of Laravel application.
   * @default 'http://localhost:80'
   */
  baseUrl: string
  /**
   * The endpoint used for channels authentication.
   * @default '/broadcasting/auth'
   */
  authEndpoint?: string
  /**
   * The endpoint used for CSRF token retrieval.
   * @default '/sanctum/csrf-cookie'
   */
  csrfEndpoint?: string
  /**
   * The name of the CSRF cookie.
   * @default 'XSRF-TOKEN'
   */
  csrfCookie?: string
  /**
   * The name of the CSRF header.
   * @default 'X-XSRF-TOKEN'
   */
  csrfHeader?: string
}

export interface ModuleOptions {
  /**
   * The Laravel Broadcasting app key for a secure connection.
   * @default undefined
   */
  key?: string
  /**
   * The Laravel broadcaster type to use.
   * @default 'reverb'
   */
  broadcaster: keyof Broadcaster
  /**
   * The host to connect to WebSocket.
   * @default 'localhost'
   */
  host?: string
  /**
   * The port to connect to WebSocket.
   * @default 8080
   */
  port?: number
  /**
   * The scheme to use for the connection.
   * @default 'https'
   */
  scheme: 'http' | 'https'
  /**
   * The application cluster to connect to.
   * @default undefined
   */
  cluster?: string
  /**
   * The transports to enable for the connection.
   * @default ['ws','wss']
   */
  transports?: string[]
  /**
   * Authentication options for Private and Presence channels.
   */
  authentication?: Authentication
  /**
   * The log level to use for the logger.
   *
   * 0: Fatal and Error
   * 1: Warnings
   * 2: Normal logs
   * 3: Informational logs
   * 4: Debug logs
   * 5: Trace logs
   *
   * More details at https://github.com/unjs/consola?tab=readme-ov-file#log-level
   * @default 3
   */
  logLevel: number
  /**
   * Additional properties to extend the Echo instance options.
   * @default undefined
   */
  properties?: object
}
